### DISCUSSION TITLE
Mollusc and Skitter genetic material rework

### DISCUSSION THREAD

**Discussion by Yoshbie** [FakeFishGames/Barotrauma#11647] [Ideas and suggestions]
The Mollusc and Skitter genes are rather lackluster and practically have no impact even at 100% condition. getting a 0.83% damage resistance or a 0.58% movement speed increase for getting hit is so negligeable it might as well not be there. 

The problem is, buffing the amount of vigor/hyperactivity gained to a usable amount also usually makes the duration of the effect last quite a while which I feel probably goes against the intended design of being a short but impactful burst of defense/mobility, similar to the hunter gene's burst of melee damage. 
My suggestion is creating a unique buff for the Mollusc and Skitter genes similar to the Hunter gene where it caps at 100 or lower affliction strength and have a high minvalue of the stattypes to be boosted (Defense for Mollusc, Movement/Swimming/Attack? Speed for Skitter) with a maxvalue not too much bigger than the latter. This way it would be the best of both worlds, having a noticeable impact like Meth and Steroids without lasting 5-10 minutes after being hit a couple times.

**Comment by Yoshbie** [comment]
Here's some code to represent the kind of idea I mean:

<Afflictions>
	<Override>
		<Affliction
		  name="Vigor On Damage"
		  identifier="vigorondamage"
		  type="geneticmaterialbuff"
		  limbspecific="false"
		  indicatorlimb="Torso"
		  hideiconafterdelay="true"
		  showiconthreshold="0"
		  treatmentthreshold="100"
		  maxstrength="100"
		  iconcolors="94,54,95;163,73,164;200,191,231">
		  <Effect minstrength="0" maxstrength="20">
			<StatusEffect target="Character" type="OnDamaged">
			  <requiredaffliction type="damage" minstrength="5" />
			  <Affliction identifier="increasedresistance_temporary" amount="20" />
			</StatusEffect>
		  </Effect>
		  <Effect minstrength="20" maxstrength="40">
			<StatusEffect target="Character" type="OnDamaged">
			  <requiredaffliction type="damage" minstrength="5" />
			  <Affliction identifier="increasedresistance_temporary" amount="40" />
			</StatusEffect>
		  </Effect>
		  <Effect minstrength="40" maxstrength="60">
			<StatusEffect target="Character" type="OnDamaged">
			  <requiredaffliction type="damage" minstrength="5" />
			  <Affliction identifier="increasedresistance_temporary" amount="60" />
			</StatusEffect>
		  </Effect>
		  <Effect minstrength="60" maxstrength="80">
			<StatusEffect target="Character" type="OnDamaged">
			  <requiredaffliction type="damage" minstrength="5" />
			  <Affliction identifier="increasedresistance_temporary" amount="80" />
			</StatusEffect>
		  </Effect>
		  <Effect minstrength="80" maxstrength="100">
			<StatusEffect target="Character" type="OnDamaged">
			  <requiredaffliction type="damage" minstrength="5" />
			  <Affliction identifier="increasedresistance_temporary" amount="100" />
			</StatusEffect>
		  </Effect>
		  <icon texture="Content/UI/TalentsIcons3.png" sheetindex="4,0" sheetelementsize="128,128" color="163,73,764,255" origin="0,0"/>
		</Affliction>
	</Override>

	  <Affliction
		name="Increased Resistance (Temporary)"
		identifier="increasedresistance_temporary"
		type="geneticmaterialbuff"
		limbspecific="false"
		indicatorlimb="Torso"
		hideiconafterdelay="true"
		showiconthreshold="0"
		treatmentthreshold="100"
		maxstrength="100"
		iconcolors="94,54,95;163,73,164;200,191,231">
		<Effect minstrength="0" maxstrength="100" strengthchange="-10"
		  resistancefor="damage"
		  minresistance="0.20"
		  maxresistance="0.50"/>
		<icon texture="Content/UI/TalentsIcons3.png" sheetindex="4,0" sheetelementsize="128,128" color="181,230,29,255" origin="0,0"/>
	  </Affliction>

	<Override>
	  <Affliction
		name="Hyperactivity On Damage"
		identifier="hyperactivityondamage"
		type="geneticmaterialbuff"
		limbspecific="false"
		indicatorlimb="Torso"
		hideiconafterdelay="true"
		showiconthreshold="0"
		treatmentthreshold="100"
		maxstrength="100"
		iconcolors="94,54,95;163,73,164;200,191,231">
		<Effect minstrength="0" maxstrength="20">
		  <StatusEffect target="Character" type="OnDamaged">
			<requiredaffliction type="damage" minstrength="5" />
			<Affliction identifier="increasedmovementspeed_temporary" amount="20" />
		  </StatusEffect>
		</Effect>
		<Effect minstrength="20" maxstrength="40">
		  <StatusEffect target="Character" type="OnDamaged">
			<requiredaffliction type="damage" minstrength="5" />
			<Affliction identifier="increasedmovementspeed_temporary" amount="40" />
		  </StatusEffect>
		</Effect>
		<Effect minstrength="40" maxstrength="60">
		  <StatusEffect target="Character" type="OnDamaged">
			<requiredaffliction type="damage" minstrength="5" />
			<Affliction identifier="increasedmovementspeed_temporary" amount="60" />
		  </StatusEffect>
		</Effect>
		<Effect minstrength="60" maxstrength="80">
		  <StatusEffect target="Character" type="OnDamaged">
			<requiredaffliction type="damage" minstrength="5" />
			<Affliction identifier="increasedmovementspeed_temporary" amount="80" />
		  </StatusEffect>
		</Effect>
		<Effect minstrength="80" maxstrength="100">
		  <StatusEffect target="Character" type="OnDamaged">
			<requiredaffliction type="damage" minstrength="5" />
			<Affliction identifier="increasedmovementspeed_temporary" amount="100" />
		  </StatusEffect>
		</Effect>
		<icon texture="Content/UI/TalentsIcons3.png" sheetindex="4,0" sheetelementsize="128,128" color="163,73,764,255" origin="0,0"/>
	  </Affliction>
	</Override>

	  <Affliction
		name="Increased Movement Speed (Temporary)"
		identifier="increasedmovementspeed_temporary"
		type="geneticmaterialbuff"
		limbspecific="false"
		indicatorlimb="Torso"
		hideiconafterdelay="true"
		showiconthreshold="0"
		treatmentthreshold="100"
		maxstrength="100"
		iconcolors="94,54,95;163,73,164;200,191,231">
		<Effect minstrength="0" maxstrength="100" strengthchange="-10">
		  <Statvalue stattype="WalkingSpeed" minvalue="0.25" maxvalue="0.5" />
		  <StatValue stattype="SwimmingSpeed" minvalue="0.25" maxvalue="0.75" />
		</Effect>
		<icon texture="Content/UI/TalentsIcons3.png" sheetindex="4,0" sheetelementsize="128,128" color="181,230,29,255" origin="0,0"/>
	  </Affliction>
</Afflictions>

**Comment by BiffW** [comment]
Yes. This is what they should be like to be consistent with hunter gene.

**Reply by BiffW** [reply]
I've just made a mod from this discussion with some tweaks.
https://steamcommunity.com/sharedfiles/filedetails/?id=2959938104
